packages/ubuntu_desktop_installer/lib/pages/who_are_you/who_are_you_page.dart

```diff
@@ -0,0 +1,368 @@
+                        controller: _usernameController,
+                        validator: MultiValidator([
+                          RequiredValidator(
+                              errorText: lang
+                                  .whoAreYouPageUsernameRequiredValidatorErrorText),
+                          MinLengthValidator(2,
+                              errorText: lang
+                                  .whoAreYouPageUsernameMinLengthValidatorErrorText),
+                          PatternValidator(
+                              r'^(?=.{2,20}$)(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+(?<![_.])$',
+                              errorText: lang
+                                  .whoAreYouPageUsernamePatternValidatorErrorText)
+                        ]),
+                        label: lang.whoAreYouPageUsernameLabel,
+                        obscureText: false,
+                        successWidget: successIcon,
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 10, bottom: 10),
+                      child: Row(
+                        children: [
+                          SizedBox(
+                            width: MediaQuery.of(context).size.width /
+                                screenFactor,
+                            child: Form(
+                              key: _passwordFormKey,
+                              child: TextFormField(
+                                autovalidateMode:
+                                    AutovalidateMode.onUserInteraction,
+                                validator: MultiValidator([
+                                  RequiredValidator(
+                                      errorText: lang
+                                          .whoAreYouPagePasswordRequiredValidatorErrorText),
+                                  MinLengthValidator(2,
+                                      errorText: lang
+                                          .whoAreYouPagePasswordMinLengthValidatorErrorText),
+                                ]),
+                                controller: _passwordController,
+                                obscureText: true,
+                                decoration: InputDecoration(
+                                    border: OutlineInputBorder(),
+                                    labelText: lang.whoAreYouPagePasswordLabel),
+                              ),
+                            ),
+                          ),
+                          ValueListenableBuilder<TextEditingValue>(
+                              valueListenable: _passwordController,
+                              builder: (context, value, child) {
+                                return Padding(
+                                    padding: const EdgeInsets.only(
+                                        left: checkMarksLeftPadding),
+                                    child: findPassWordStrengthLabel(
```

Comment:
can we make `ValidatedInput` more generic in the sense that it can build any indicator widget. not just the tickmark indicating success, but also the password strength label? then we can use `ValidatedInput` for all fields. it could be `ValueWidgetCallback` instead of a `Widget` and pass the valid state as a value. then you can build the appropriate strength label here